### ISSUE TITLE
Unnecessary stack elements when SelectQueryImpl doesn't have both semi join and where clause predicates

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10548]
I've noticed some repeated occurrence of `ConditionProviderImpl` on the stack trace of a query execution, see indented code:

```
RowSubqueryCondition.accept(Context<?>) line: 135	
DefaultRenderContext.visit0(QueryPartInternal) line: 609	
DefaultRenderContext(AbstractContext<C>).visit(QueryPart) line: 264	
CompareCondition.accept(Context<?>) line: 127	
DefaultRenderContext.visit0(QueryPartInternal) line: 609	
DefaultRenderContext(AbstractContext<C>).visit(QueryPart) line: 264	
CombinedCondition.accept(Context<?>) line: 184	
DefaultRenderContext.visit0(QueryPartInternal) line: 609	
DefaultRenderContext(AbstractContext<C>).visit(QueryPart) line: 264	
    ConditionProviderImpl.accept(Context<?>) line: 124	
    DefaultRenderContext.visit0(QueryPartInternal) line: 609	
    DefaultRenderContext(AbstractContext<C>).visit(QueryPart) line: 264	
    ConditionProviderImpl.accept(Context<?>) line: 124
    DefaultRenderContext.visit0(QueryPartInternal) line: 609	
    DefaultRenderContext(AbstractContext<C>).visit(QueryPart) line: 264	
SelectQueryImpl<R>.toSQLReference0(Context<?>, Field<?>[], Field<?>[]) line: 1730	
SelectQueryImpl<R>.toSQLReferenceLimitDefault(Context<?>, Field<?>[], Field<?>[]) line: 1249	
SelectQueryImpl<R>.accept0(Context<?>) line: 1141	
SelectQueryImpl<R>.accept(Context<?>) line: 916	
DefaultRenderContext.visit0(QueryPartInternal) line: 609	
DefaultRenderContext(AbstractContext<C>).visit(QueryPart) line: 264	
SelectQueryImpl<R>(AbstractQuery).getSQL0(ExecuteContext) line: 524	
SelectQueryImpl<R>(AbstractQuery).execute() line: 326	
SelectQueryImpl<R>(AbstractResultQuery<R>).fetchLazy(int) line: 481	
SelectQueryImpl<R>(AbstractResultQuery<R>).fetchLazy() line: 450	
SelectQueryImpl<R>(AbstractResultQuery<R>).fetchLazyNonAutoClosing() line: 464	
SelectQueryImpl<R>(AbstractResultQuery<R>).fetchOne() line: 641	
SelectImpl<R,T1,T2,T3,T4,T5,T6,T7,T8,T9,T10,T11,T12,T13,T14,T15,T16,T17,T18,T19,T20,T21,T22>.fetchOne() line: 3074	
```

This is because of the following logic in `SelectQueryImpl`:

```java
ConditionProviderImpl actual = new ConditionProviderImpl();

if (semiAntiJoinPredicates != null)
    actual.addConditions(semiAntiJoinPredicates);

if (where.hasWhere())
    actual.addConditions(where);

context.formatSeparator()
       .visit(K_WHERE)
       .sql(' ')
       .visit(actual);
```

We don't have to wrap the `where` clause in another `ConditionProviderImpl`, *unless* we also have a semi join predicate. Not a huge improvement, but still, almost all queries are affected.

**Comment by lukaseder** [comment]
The fix is simply:

```java
if (where.hasWhere())
    actual.addConditions(where.getWhere());
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.5 (#10549)
